### PR TITLE
add OmniDataAdapter

### PR DESCRIPTION
**Pull request by Jutao-liu**

### CODE REVIEW

**Review by xulilililili — CHANGES_REQUESTED** [review]
If the omnidata server node fails to be pushed down, you are advised to print the host name of the failed node

**Review by xulilililili — CHANGES_REQUESTED** [review]
`int retryTime = Math.min(4, omniDataHosts.size());`
You are advised to use the final constant instead.

**Review by xulilililili — APPROVED** [review]
good

**Comment by Jutao-liu** [comment]
> `int retryTime = Math.min(4, omniDataHosts.size());` You are advised to use the final constant instead.

Good advice. Completed modification

File: connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/reader/OmniDataAdapter.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            ColumnVector[] batchVector = batchVectors.poll();
+            // channelCount: column, positionCount: row
+            int channelCount = batchVector.length;
+            int positionCount = batchVector[0].isNull.length;
+            if (ndpPredicateInfo.getIsPushDownAgg()) {
+                // agg raw return
+                System.arraycopy(batchVector, 0, batch.cols, 0, channelCount);
+            } else {
+                for (int i = 0; i < channelCount; i++) {
+                    int columnId = ndpPredicateInfo.getOutputColumns().get(i);
```
Comment (tedliUCAS) [inline connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/reader/OmniDataAdapter.java:151]: get(i)存在空指针异常风险

File: connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/reader/OmniDataAdapter.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                columnTypes[index] = OmniDataUtils.transOmniDataDecodeType(codeType);
+            }
+        }
+
+        PageDeserializer deserializer = new PageDeserializer(columnTypes);
+
+        Queue<ColumnVector[]> pages = new LinkedList<>();
+        int failedTimes = 0;
+        for (String omniDataHost : omniDataHosts) {
+            omniDataProperty.getProperties().put("grpc.client.target", omniDataHost + ":" + omniDataProperty.getPort());
```
Comment (tedliUCAS) [inline connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/reader/OmniDataAdapter.java:76]: 适配HAF的场景不需要端口号配置

File: connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/reader/OmniDataAdapter.java (hunk 1 above)
Comment (Jutao-liu) [reply connector/src/main/java/org/apache/hadoop/hive/ql/omnidata/reader/OmniDataAdapter.java:151]: Good advice. Completed modification

**Pull request merged**